resp/get_foxsi_optics_effarea.pro

```diff
@@ -22,29 +24,31 @@ IF n_elements(offaxis_angle) EQ 1 THEN angle = 1/sqrt(2) * [offaxis_angle, offax
 ; Switch X0->D6 and X6->D0 - we do not need this now that we follow the optic number convention
 ;if MODULE_NUMBER eq 0 then MODULE = 6 else if MODULE_NUMBER eq 6 then MODULE = 0 $
 ;		else MODULE = MODULE_NUMBER
-		
-files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI2_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
-         'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt']
 
-; todo: these need to be provided by the data files themselves
-angles = [-9, -7, -5, -3, -1, 0, 1, 3, 5, 7, 9]
-energy = [4.5,  5.5,  6.5,  7.5,  8.5,  9.5, 11. , 13. , 15. , 17. , 19. , 22.5, 27.5]
+
+CASE DATE OF
```

Comment:
I also thought a variable YEAR would be more "intuitive". However, it is not defined inside the COMMON FOXSI variable (see init_param.pro for details). I see three possible ways we can proceed in this respect:

1. Use the COMMON variable DATE which is unique for each flight and set when you run "FOXSI, YEAR" on the SSWIDL prompt. The advantage of this method is that we do not need to include any additional mandatory keyword into the get_foxsi_optics_effarea function that would break other scripts already existing that use this routine without that YEAR additional keyword.

2. Add the variable YEAR to the Init_param and call YEAR instead of DATE inside get_foxsi_optics_effarea. The only advantage of this solution is to make the script more intuitive but it would not be more or less functional than what I proposed already. 

3. Use the DATE variable in which case we would minimize the number of other files needed to be modified. 

What do you think? :)

Reply:
This is a very good point. I think I still prefer option 1, using the YEAR keyword, for the following reason: 
- I think this routine could be used without starting the FOXSI soft with init_param.pro. For example if someone wants to plot the different effective areas from the past three flights, the presence of the YEAR keyword. Inside the function, you could have something like
IF date is defined, then use date to set the default value of the YEAR keyword
ELSE use a default value such as 2014... I think FOXSI2 is the default value in the software now.
What do you think?

Reply:
I think we should try to avoid making one FOXSI flight (FOXSI-2) the standard flight. Instead, I think all parameters and variables should be consistent with what the user loads when running: IDL> foxsi, year

I think the way the code works now allows a user to easily plot the EA for the last three flights by loading foxsi three times. Something like:

IDL> foxsi, 2012
IDL> EA1 = get_foxsi_optics_effarea(...)
IDL> foxsi, 2014
IDL> EA2 = get_foxsi_optics_effarea(...)
IDL> foxsi, 2018
IDL> EA3 = get_foxsi_optics_effarea(...)

and then plot EA1, EA2, and EA3. I know this requires loading foxsi three times which is not optimal, but I think it's a fair payoff to preserve consistency among flight variables, and avoids possible mistakes where a user could accidentally use an EA from FOXSI-2 when for example analyzing FOXSI-3 data (a more probable scenario).